### PR TITLE
Fix dark mode detection on SSR to avoid blinking 

### PR DESCRIPTION
**Pull request by odinuge**
This ensures that the correct theme is applied _before_ the SSR content is show to the user. That removes the blinking for users using the auto theme and are using dark mode.

https://user-images.githubusercontent.com/1467188/166219820-aab09c44-d834-4803-bc3b-74285f9e808d.mp4

PS: This also adds 'auto' as the selected theme for unauthenticated users.

### CODE REVIEW

File: server/pageRenderer.js [hunk 1]
```diff
@@ -120,6 +118,21 @@ export default function pageRenderer({
...
       </head>
       <body>
+        <script>
+          ${
+            // If user has selected auto and device is in dark mode; ensure we update
+            // the theme before first render to screen
+            selectedTheme === 'auto'
+              ? `(function () {
+            try {
+              if (window.matchMedia('(prefers-color-scheme: dark)').matches === true) {
```
Comment (odinuge) [inline server/pageRenderer.js:128]: It's also possible to subscribe to changes here, but that is outside the scope here, and others can feel free to do that :ok_hand:

File: server/pageRenderer.js [hunk 2]
```diff
@@ -120,6 +118,21 @@ export default function pageRenderer({
...
+                document.documentElement.setAttribute('data-theme', 'dark');
+              }
+            } catch (e) {}`
+              : ''
+          }
+          })();
+        </script>
```
Comment (LudvigHz) [inline server/pageRenderer.js:135]: Might it be an idea to put the whole `script` tag in the ternary, so we don't end up with an empty tag that might confuse people?

**Review by LudvigHz — APPROVED** [review]
LGTM :shipit:

File: server/pageRenderer.js (hunk 2 above)
Comment (odinuge) [reply server/pageRenderer.js]: Skal være good nå

**Review by LudvigHz — APPROVED** [review]

**Pull request merged**